pkg/segment/results/segresults/segresults_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package segresults
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_Remote_Stats(t *testing.T) {
+
+	evalStatsMetaData := EvalStatsMetaData{
+		RangeStat: &structs.RangeStat{
+			Min: -1,
+			Max: 1,
+		},
+		AvgStat: &structs.AvgStat{
+			Sum:   10,
+			Count: 2,
+		},
+		StrSet: map[string]struct{}{
+			"test":  {},
+			"test2": {},
+		},
+		StrList:       []string{"abc", "def"},
+		MeasureResult: 123,
+	}
+
+	myNums := structs.NumericStats{
+		Min: utils.NumTypeEnclosure{Ntype: utils.SS_DT_SIGNED_NUM,
+			IntgrVal: 456},
+		Max: utils.NumTypeEnclosure{Ntype: utils.SS_DT_FLOAT,
+			FloatVal: 23.4567},
+		Sum: utils.NumTypeEnclosure{Ntype: utils.SS_DT_SIGNED_NUM,
+			IntgrVal: 789},
+		Dtype: utils.SS_DT_USIGNED_16_NUM,
+	}
+
+	segStat := structs.SegStats{
+		IsNumeric: true,
+		Count:     2345,
+		NumStats:  &myNums,
+		StringStats: &structs.StringStats{
+			StrSet: map[string]struct{}{
+				"str1": {},
+				"str2": {},
+			},
+			StrList: []string{
+				"str1",
+				"str2",
+			},
+		},
+	}
+	segStat.CreateNewHll()
+
+	for i := 0; i < 3200; i++ {
+		segStat.InsertIntoHll([]byte(fmt.Sprintf("mystr:%v", i)))
+	}
+
+	evalStats := map[string]EvalStatsMetaData{
+		"eval1": evalStatsMetaData,
+	}
+
+	remoteStats := &RemoteStats{
+		EvalStats: evalStats,
+		SegStats:  []*structs.SegStats{&segStat},
+	}
+
+	remoteStatsJson, err := remoteStats.RemoteStatsToJSON()
+	assert.Nil(t, err)
+	assert.NotNil(t, remoteStatsJson)
+
+	remoteStats2, err := remoteStatsJson.ToRemoteStats()
+	assert.Nil(t, err)
+	assert.NotNil(t, remoteStats2)
+
+	assert.Equal(t, evalStats, remoteStats.EvalStats)
+	assert.Equal(t, segStat, *remoteStats.SegStats[0])
```

Comment:
A few things:
1. For lines 96-97, it looks like you're not really testing anything. You set those directly on 84-85. Am I missing something?
2. Lines 88-94, you're doing pretty light testing. Yes, we expect no errors and a non-nil result, but actually we expect a specific non-nil result; you should test that—at least for `remoteStats2`
3. `remoteStats2` isn't very clear. The test is that `remoteStats` and `remoteStats2` should be the same, because you essentially encoded and then decoded `remoteStats`; you should rename `remoteStats2`

Reply:
addressed in the latest commit.